data_safe_haven/allowlist/allowlist.py

```diff
@@ -0,0 +1,120 @@
+from difflib import unified_diff
+from typing import Self
+
+from data_safe_haven.config import Context
+from data_safe_haven.external import AzureSdk
+from data_safe_haven.types import AllowlistRepository
+
+
+class Allowlist:
+    """Allowlist for packages"""
+
+    @classmethod
+    def from_remote(
+        cls: type[Self],
+        context: Context,
+        *,
+        repository: AllowlistRepository,
+        sre_resource_group: str,
+        storage_account_name: str,
+    ) -> str:
+        """Get the current package allowlist"""
+
+        # Get the Azure SDK
+        azure_sdk = AzureSdk(subscription_name=context.subscription_name)
+
+        # Get the file share name
+        file_share_name = "software-repositories-nexus-allowlists"
```

Comment:
It would be nice to do this in a way that we ensure it is the same as what Pulumi will deploy.